Yoast/wordpress-seo
# Add translations for custom components

## Summary

This PR can be summarized in the following changelog entry:
Adds the possibility for the strings/text in the custom components to be translated.

## Relevant technical choices:
- Uses i18n from yoast-components.
- The injected translate function is set as a required function propType for the components.

## Test instructions

This PR can be tested by following these steps:
1. Start the wizard and see if the text in the first step (Newsletter sign-up) shows correctly.
2. Go to step 8 google search console and see if the text is correct.
3. Go to step 9 title settings and see if this text is correct.
NOTE: Also play with the components a bit to see if nothing is broken and all text is O.K.

Fixes #5666
Fixes #5642
Fixes #5590
